lib/bureaucrat/api_blueprint_writer.ex

```diff
@@ -0,0 +1,159 @@
+defmodule Bureaucrat.ApiBlueprintWriter do
+  def write(records, path) do
+    file = File.open!(path, [:write, :utf8])
+    records = group_records(records)
+    title = Application.get_env(:bureaucrat, :title)
+    puts(file, "# #{title}\n\n")
+    write_api_doc(records, file)
+  end
+
+  defp write_api_doc(records, file) do
+    Enum.each(records, fn {controller, actions} ->
+      %{request_path: path} = Enum.at(actions, 0) |> elem(1) |> List.first()
+      puts(file, "\n# Group #{controller}")
+      puts(file, "## #{controller} [#{path}]")
+
+      Enum.each(actions, fn {action, records} ->
+        write_action(action, controller, Enum.reverse(records), file)
+      end)
+    end)
+
+    puts(file, "")
+  end
+
+  defp write_action(action, controller, records, file) do
+    test_description = "#{controller} #{action}"
+    record_request = Enum.at(records, 0)
+    method = record_request.method
+    path_params = record_request.path_params
+
+    file |> puts("### #{test_description} [#{method} #{anchor(record_request)}]")
+
+    unless path_params == %{} do
```

Comment:
can we extract this to function, and maybe use pattern matching we can remove unless. Like,
```elixir
defp write_paramters(_path_params =%{}, file) do
  nil
end
defp write_parameters(path_params, file)
file
     |> puts("\n+ Parameters\n#{formatted_params(path_params)}")

      Enum.each(path_params, fn {param, value} ->
        puts(file, indent_lines(12, "#{param}: #{value}"))
      end)
end
```